public/app/core/components/code_editor/code_editor.ts

```diff
@@ -159,7 +163,7 @@ function link(scope, elem, attrs) {
 
   function setLangMode(lang) {
     let aceModeName = `ace/mode/${lang}`;
-    setModuleUrl("mode", lang);
+    setModuleUrl("mode", lang, scope.$parent.datasource.meta.baseUrl || null);
```

Comment:
not good to access scope parent like this, very brittle (easy to break), better to add a binding property on the directive. If we make this change we should move the prometheus lang file so that it is under the prometheus plugin dir.